src/solvers/smt2_incremental/smt_bit_vector_theory.cpp

```diff
@@ -42,3 +42,152 @@ static void validate_bit_vector_predicate_arguments(
     smt_bit_vector_theoryt::the_name{};
 #include "smt_bit_vector_theory.def"
 #undef SMT_BITVECTOR_THEORY_PREDICATE
+
+const char *smt_bit_vector_theoryt::additiont::identifier()
+{
+  return "bvadd";
+}
+
+smt_sortt smt_bit_vector_theoryt::additiont::return_sort(
+  const smt_termt &lhs,
+  const smt_termt &rhs)
+{
+  // For now, make sure that they have the same bit-width
+  additiont::validate(lhs, rhs);
+  const auto width = lhs.get_sort().cast<smt_bit_vector_sortt>()->bit_width();
+  return smt_bit_vector_sortt{width};
+}
+
+void smt_bit_vector_theoryt::additiont::validate(
+  const smt_termt &lhs,
+  const smt_termt &rhs)
+{
+  validate_bit_vector_predicate_arguments(lhs, rhs);
+}
+
+const smt_function_application_termt::factoryt<
+  smt_bit_vector_theoryt::additiont>
+  smt_bit_vector_theoryt::addition{};
+
+const char *smt_bit_vector_theoryt::subtractiont::identifier()
+{
+  return "bvsub";
```

Comment:
❗ `bvsub` does not appear to be a valid operator in bit vector theory, according to this definition - https://smtlib.cs.uiowa.edu/theories-FixedSizeBitVectors.shtml . We should probably exclude it from our definition of the theory and convert minus expressions to a combination of plus and negation terms instead.